itest/rust/src/object_tests/object_test.rs

```diff
@@ -1076,3 +1076,59 @@ fn double_use_reference() {
     double_use.free();
     emitter.free();
 }
+
```

Comment:
This isn't really testing objects (`Gd`) but rather the registration API (`godot::register::godot_api` macro).

As such, it should be in [`register_tests`](https://github.com/godot-rust/gdext/tree/master/itest/rust/src/register_tests) directory. Maybe you could even crate a new file `multiple_impls_test.rs`?